ghcide/src/Development/IDE/Main.hs

```diff
@@ -232,6 +231,7 @@ data Arguments = Arguments
     , argsHandleIn              :: IO Handle
     , argsHandleOut             :: IO Handle
     , argsThreads               :: Maybe Natural
+    , argsLspEnvCb              :: LSP.LanguageContextEnv Config -> IO () -- ^ Callback to run once the LSP environment is setup
```

Comment:
This is a bit open ended - at what time is the callback run? Importantly, is it before or after LSP initialisation?

Reply:
At the moment it only promises to be called when we have a `LanguageContextEnv`, but in fact it will be called during initialization. I agree it's not terribly precise, but I don't think representing it as a response to a LSP message is much more precise: it's not really a message handler!

Reply:
It is important to be more precise because the LSP server is restricted in what it can do before initialisation. So to be safe, I think the callback should only be called **after** LSP initialisation is complete

Reply:
You know what, maybe you were just right to begin with.
- A sensible time to do this _is_ when you get `Initialized`
- The way that we have to add notification handlers is plugins, so adding a plugin ad hoc like this is maybe not crazy

Reply:
I tend to agree 😊